utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,218 @@
+// Source C-CDA to FHIR - Datatype Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerinax/health.fhir.r4;
+
+xmlns "urn:hl7-org:v3" as v3;
+
+public isolated function mapCcdaIdToFhirIdentifier(xml<xml:Element> idElement) returns r4:Identifier|error {
+    if idElement is xml:Element && idElement.length() > 0 {
+        string|error idVal = idElement.extension;
+        string|error rootVal = idElement.root;
+        r4:Identifier identifier = {};
+        if rootVal is string {
+            if rootVal.equalsIgnoreCaseAscii("2.16.840.1.113883.4.1") {
+                identifier.system = "http://hl7.org/fhir/sid/us-ssn";
+                identifier.value = check idVal;
+            } else if idVal is string {
+                identifier.system = rootVal;
+                identifier.value = string `urn:oid:${idVal}`;
+            } else {
+                identifier.system = "urn:ietf:rfc:3986";
+                identifier.value = rootVal;
+            }
+            return identifier;
+        }
+    }
+    return {};
+}
+
+public isolated function mapCcdaAddressToFhirAddress(xml<xml:Element> addressElement, boolean isNamespaceAvailable = true) returns r4:Address|error? {
+    if addressElement is xml:Element && addressElement.length() > 0 {
+        r4:Address address = {};
+        xml<xml:Element> streetAddressLineElement;
+        xml<xml:Element> cityElement;
+        xml<xml:Element> stateElement;
+        xml<xml:Element> postalCodeElement;
+
+        if isNamespaceAvailable {
+            streetAddressLineElement = addressElement.children().<v3:streetAddressLine>;
+            cityElement = addressElement.children().<v3:city>;
+            stateElement = addressElement.children().<v3:state>;
+            postalCodeElement = addressElement.children().<v3:postalCode>;
+        } else {
+            streetAddressLineElement = addressElement.children().<streetAddressLine>;
+            cityElement = addressElement.children().<city>;
+            stateElement = addressElement.children().<state>;
+            postalCodeElement = addressElement.children().<postalCode>;
+        }
+
+        if streetAddressLineElement is xml:Element && streetAddressLineElement.length() > 0 {
+            address.line = [streetAddressLineElement.get(0).data()];
+        }
+
+        if cityElement is xml:Element && cityElement.length() > 0 {
+            address.city = cityElement.get(0).data();
+        }
+
+        if stateElement is xml:Element && stateElement.length() > 0 {
+            address.state = stateElement.get(0).data();
+        }
+
+        if postalCodeElement is xml:Element && postalCodeElement.length() > 0 {
+            address.postalCode = postalCodeElement.get(0).data();
+        }
+        return address;
+    }
+    return;
+}
+
+public isolated function mapCcdaTelecomToFhirTelecom(xml<xml:Element> telecomElement) returns r4:ContactPoint|error? {
```

Comment:
This function never returns an `error?`

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/bd15548be31a7e76b4962973508cdedd6f8fd58a